style_editor.html

```diff
@@ -9280,37 +9494,220 @@
 
 function ClickLockup() {
   STATE_LOCKUP = STATE_LOCKUP == LOCKUP_NORMAL ? 0 : LOCKUP_NORMAL;
-  UpdateLockupButtons();
   blade.addEffect(STATE_LOCKUP ? EFFECT_LOCKUP_BEGIN : EFFECT_LOCKUP_END, Math.random() * 0.7 + 0.2);
 }
 
 function ClickDrag() {
   STATE_LOCKUP = STATE_LOCKUP == LOCKUP_DRAG ? 0 : LOCKUP_DRAG;
-  UpdateLockupButtons();
   blade.addEffect(STATE_LOCKUP ? EFFECT_DRAG_BEGIN : EFFECT_DRAG_END, 1.0);
 }
 
 function ClickLB() {
   STATE_LOCKUP = STATE_LOCKUP == LOCKUP_LIGHTNING_BLOCK ? 0 : LOCKUP_LIGHTNING_BLOCK;
-  UpdateLockupButtons();
 }
 
 function ClickMelt() {
   STATE_LOCKUP = STATE_LOCKUP == LOCKUP_MELT ? 0 : LOCKUP_MELT;
-  UpdateLockupButtons();
 }
 
-function ClickInhilt() {
-  STATE_NUM_LEDS = 145 - STATE_NUM_LEDS;
-  var button = FIND("INHILT_BUTTON");
-  button.style.backgroundColor = STATE_NUM_LEDS == 1 ? "dodgerblue" : "";
-}
+/* Settings buttons saved as local cookies */
+var darkOn = true;
+var tipsOn = true;
+var stillOn = false;
+var swingEmOn = true;
+var inhiltOn = false;
+var slowOn = false;
+/* var use_graflex = true; is already initialized above */
 
-function ClickSlow() {
-  framesPerUpdate = 60 - framesPerUpdate;
-  var button = FIND("SLOW_BUTTON");
-  button.style.backgroundColor = framesPerUpdate == 60 ? "dodgerblue" : "";
+function getSavedCookie(buttonState) {
+  var value = localStorage.getItem(buttonState);
+  if (buttonState === "wavlenValue") {
+    return parseInt(value);
+  }
+  return value;
 }
+function saveCookie(buttonState, settingIsOn) {
+  localStorage.setItem(buttonState, settingIsOn);
+}
+
+// Call the onPageLoad function when the page is loaded
+window.addEventListener('load', onPageLoad);
+
+var myWavLen = new WavLenClass();
+var wavlenValue;
+var variantTimeoutId;
+
+function onPageLoad() {
+  // Retrieve saved cookies and set buttons and settings accordingly
+  var body = document.querySelector("body");
+
+  // DARK_BUTTON
+  var darkButton = FIND("DARK_BUTTON");
+  var darkState = getSavedCookie("darkState");
+  var structuredView = FIND("structured_view")
+  // console.log("BC-getSavedCookie called, darkState = " + darkState);
+  if (darkState !== "false") {
```

Comment:
There is a lot of duplication between onPageLoad() and ClickDark() (and other click functions)
Seems like we should have something like an SetDarkmode(onoff) function which can be called from both places.